resources/test-invoker.mjs

```diff
@@ -48,8 +67,51 @@ export class RAFTestInvoker extends TestInvoker {
     }
 }
 
+class AsyncRAFTestInvoker extends BaseRAFTestInvoker {
+    _scheduleCallbacks(resolve) {
+        requestAnimationFrame(async () => {
+            await this._syncCallback();
+
+            let gotTimer = false;
+            let gotMessage = false;
+
+            const tryTriggerAsyncCallback = () => {
+                if (!gotTimer || !gotMessage)
+                    return;
+
+                this._asyncCallback();
+                setTimeout(async () => {
+                    await this._reportCallback();
+                    resolve();
+                }, 0);
+            };
+
+            setTimeout(() => {
+                gotTimer = true;
+                tryTriggerAsyncCallback();
+            });
```

Comment:
oh, hmm, does this non-double rAF setup have a bug after all.  If the promise callbacks create new promises which get resolve and those callbacks then trigger setTimeout or postMessage, does anything guarantee that test runners setTimeout/messageport gets handled after those?

Reply:
We are stopping and waiting for `await this._syncCallback();`. 
The setTimeout / messageport happens after that resolved, correct?
The reason of having the boolean flags is that we can't guarantee in which order setTimeout / postMessage gets called. At least that's how I understood our conversation 🙂

Reply:
I did test it with the current tests as well as with a test that had an artificial promise resolving after a long period of time in each step.

Reply:
But await this._syncCallback(); might return in the same task, if one just does effectively
return new Promise(r => r());

In my example using MessageChannel there were still two rAFs.

I thought we were thinking to use the same runner for async tests and non-async, and I'm worried that not having two rafs would broke non-async tests (meaning that some time wouldn't be measured correctly).  But I could be missing something (I think I did when I said on Slack couple of days ago that this approach is ok :)).

Reply:
@smaug---- following up on the above snippet. Do you think this would satisfy async steps with two rafs?